### ISSUE TITLE
Add support for resolving container image digests by tag name

### ISSUE THREAD

**Issue by ramaro** [deepmind/kapitan#142]

**Comment by uberspot** [comment]
@ramaro Can we add some more context/info and tags here?

**Comment by yoshi-1224** [comment]
I've been exploring this topic lately, and am wondering about:

1. how we should identify whether the user actually means to write a docker tag in a file or not, that is, if any pieces of text that resembles `foo:bar` format may not necessarily refer to docker tag. Maybe we intend to limit this feature only to `FROM` line in Dockerfile?

2. How to get sha256 digest from docker tags. There are docker APIs v2 (run by Docker daemon), but I don't actually know if we can take it for granted that this service is locally/remotely available. Another way of course if to run some `docker pull` (or `docker images --digests`, if the tag was already used) command behind the scene and parse the digest, but I feel like `pull` is a bit of overkill. Am I missing a Docker feature that you have on mind to accomplish this task?

**Comment by uberspot** [comment]
I have the impression that this could actually be easier to solve if #197 was done. Then you could get the latest digest with a custom command at compile time.

**Comment by yoshi-1224** [comment]
@uberspot 
Agreed! That way we can support more commands, and let the user decide whether to use tags or digests to start with, as well as where to get the digest from. I'll look into #197 then.

**Comment by ademariag** [comment]
ping ?

**Comment by gburiola** [comment]
I can see what the problem is, but I don't think kapitan should be doing anything about it.

If a user really specified `image:latest` then I don't think kapitan should magically replace that with `image:sha_of_latest_image_currently_available`

The answer to that depends on each user deployment process and should probably be solved in CI/CD pipelines or with the image building process and appropriate tagging.

If we want to go ahead with this, then we should still continue to support all 3 use cases:

* use label and never change it
* use sha and never change it
* use label and replace it with sha at compile time

**Comment by ademariag** [comment]
we could use https://get-kbld.io/ instead
